aten/src/ATen/core/ivalue.h

```diff
@@ -573,25 +573,13 @@ struct C10_EXPORT ivalue::Future final : c10::intrusive_ptr_target {
   * Wait on the future until it completes.
   */
   void wait() {
+    std::unique_lock<std::mutex> lock(mutex_);
```

Comment:
If we lock before checking `completed()`, then completed probably doesn't need to itself be a atomic_bool

Reply:
Per offline discussion, we're keeping the atomic_bool and leveraging it to get fast-fallthrough on the already-completed case.